index.js

```diff
@@ -141,33 +204,47 @@ const line = []
 
 
 // *** Uncomment the lines below to test
-// console.log("%cScope & Closures - Question 1", "color: red")
+console.log("%cScope & Closures - Question 1", "color: red")
 
-// console.log(takeATicketNumber(line))
-// // => `Welcome. You are ticket number 1`
+let ticket = 0
 
-// console.log(takeATicketNumber(line))
-// // => `Welcome. You are ticket number 2`
+function takeATicketNumber(line){
+  ticket += 1
+  line.push(ticket)
+
+  return `Welcome, You are ticket number ${ticket}`
+}
+console.log(takeATicketNumber(line))
+// => `Welcome. You are ticket number 1`
 
-// console.log(nowServing(line))
+console.log(takeATicketNumber(line))
+// => `Welcome. You are ticket number 2`
+
+console.log(nowServing(line))
 // // => `Currently serving 1.`
 
-// console.log(nowServing(line))
+console.log(nowServing(line))
 // // => `Currently serving 2.`
 
-// console.log(takeATicketNumber(line))
+console.log(takeATicketNumber(line))
 // // => `Welcome. You are ticket number 3`
 
 // console.log("%c----------", "color: red") 
 
 // ***** Scope & Closures - Question 2 *****
 // *** Uncomment the lines below to test
-// console.log("%cScope & Closures - Question 2", "color: red")
+console.log("%cScope & Closures - Question 2", "color: red")
 
-// const newLine = []
-// const takeATicketNumberFunc = ticketNumberGeneratorFunc()
-// console.log(takeATicketNumberFunc(newLine))
-// // => `Welcome. You are ticket number 1`
+function ticketNumberGeneratorFunc(takeATicketNumber){
+  ticket = 0
+  return takeATicketNumber(line)
```

Comment:
In JS, functions are considered to be objects, so function definitions can be returned from a function invocation!
So your solution might look something like this:
```
function ticketNumberGeneratorFunc(){
   // ...
   return function(){
   }
}
```